server.js

```diff
@@ -8,12 +9,63 @@ app.use(cors());
 
 //Use this array as your (in-memory) data store.
 const bookings = require("./bookings.json");
+const { request, response } = require("express");
 
 app.get("/", function (request, response) {
   response.send("Hotel booking server.  Ask for /bookings, etc.");
 });
 
-// TODO add your routes and helper functions here
+app.get("/bookings", (request, response) => {
+  response.json({ bookings });
+});
+
+app.get("/bookings/:id", (request, response) => {
+  const id = Number(request.params.id);
+  const findId = bookings.find((bookings) => bookings.id === id);
+  if (!findId) {
+    return response
+      .status(404)
+      .json({ message: `booking with the ${id} not found` });
+  }
+  response.status(200).json({ findId })
+});
+
+app.post("/bookings", (request, response) => {
+  if (
+    request.body.firstName === "" ||
+    request.body.email === "" ||
+    request.body.surname == "" ||
+    request.body.title === "" ||
+    request.body.checkInDate === "" ||
+    request.body.checkOutDate === "" ||
+    request.body.roomId === null
+  ) {
+    response.status(400).json({ message: "please fill all the fields" });
+    return;
+  }
+  const newBooking = {
+    id: bookings.length + 1,
+    title: request.body.title,
+    firstName: request.body.firstName,
+    surname: request.body.surname,
+    email: request.body.email,
+    roomId: request.body.roomId,
+    checkInDate: request.body.checkInDate,
+    checkOutDate: request.body.checkOutDate,
+  };
+  bookings.push(newBooking);
+  response.status(201).json({ bookings });
+});
+
+app.delete("/bookings/:id", (request, response) => {
+  const requestId = Number(request.params.id);
+  let result = bookings.filter((item) => item.id !== requestId);
+  if (requestId < 0) {
+    return response.status(404).json({ msg: "message not found" });
+
+  }
```

Comment:
this isn't working as expected at the moment - if I put in an id that doesn't exist (that is larger or equal to 0)  I don't get the 404 - have a think what array method could be used here to check if the id exists in the bookings array.
Also really good that you're filtering the bookings so you end up with an array without the given id, but at the moment you're just sending the new array back in the response, remember to actually update the bookings array :)